clio-integration-test/src/it/scala/org/broadinstitute/clio/integrationtest/RecoveryIntegrationSpec.scala

```diff
@@ -37,50 +49,71 @@ class RecoveryIntegrationSpec
   private val initUbams = Seq.tabulate(documentCount) { i =>
     val flowcellBarcode = s"flowcell$randomId"
     val libraryName = s"library$randomId"
-    DocumentWgsUbam(
-      upsertId = UpsertId.nextId(),
-      entityId = Symbol(s"$flowcellBarcode.$i.$libraryName.${location.entryName}"),
+    val keyJson = TransferUbamV1Key(
       flowcellBarcode = flowcellBarcode,
       lane = i,
       libraryName = libraryName,
-      location = location,
+      location = location
+    ).asJson
+    val metadataJson = TransferUbamV1Metadata(
       ubamPath = Some(randomUri(i)),
       documentStatus = Some(DocumentStatus.Normal)
-    ).asJson(ElasticsearchIndex.WgsUbam.encoder)
+    ).asJson
+    val upsertIdJson = Map(ElasticsearchIndex.UpsertIdElasticsearchName -> UpsertId.nextId()).asJson
+    val entityIdJson = Map(
+      ElasticsearchIndex.EntityIdElasticsearchName -> Symbol(
+        s"$flowcellBarcode.$i.$libraryName.${location.entryName}"
+      )
+    ).asJson
+    keyJson.deepMerge(metadataJson).deepMerge(upsertIdJson).deepMerge(entityIdJson)
```

Comment:
This is pretty much the body of `ElasticsearchDocumentMapper`. Could we move that class into the `dataaccess-model` project, so we can reuse its logic here?

Reply:
Same comments for gvcf and cram setup below.